docs/API.md

```diff
@@ -104,132 +105,161 @@ gulp.task('somename', function() {
 
 #### name
 
-The name of the task. Tasks that you want to run from the command line should not have spaces in them.
+Optional, The name of the task. Tasks that you want to run from the command line 
+should not have spaces in them.
 
-#### deps
-Type: `Array`
+If the name is not provided, the task will be name after the function `name` attribute.
 
-An array of tasks to be executed and completed before your task will run.
+Note that ES 6 `Function.name` is not writable. It cannot be set or edited:
 
 ```js
-gulp.task('mytask', ['array', 'of', 'task', 'names'], function() {
-  // Do stuff
-});
+// foo.name === 'foo'
+function foo() {};
+
+// bar.name === ''
+var bar = function() {};
+
+// bar.name === ''
+bar.name = 'bar'
 ```
 
-**Note:** Are your tasks running before the dependencies are complete?  Make sure your dependency tasks are correctly using the async run hints: take in a callback or return a promise or event stream.
+You should either provide the task name or avoid anonymous functions.
 
 #### fn
 
-The function that performs the task's operations. Generally this takes the form of `gulp.src().pipe(someplugin())`.
+The function that performs the task's operations. 
+Generally this takes the form of `gulp.src().pipe(someplugin())`.
 
 #### Async task support
 
 Tasks can be made asynchronous if its `fn` does one of the following:
 
-##### Accept a callback
-
-```javascript
-// run a command in a shell
-var exec = require('child_process').exec;
-gulp.task('jekyll', function(cb) {
-  // build Jekyll
-  exec('jekyll build', function(err) {
-    if (err) return cb(err); // return error
-    cb(); // finished task
-  });
-});
-```
-
 ##### Return a stream
 
 ```js
 gulp.task('somename', function() {
-  var stream = gulp.src('client/**/*.js')
+  return gulp.src('client/**/*.js')
     .pipe(minify())
     .pipe(gulp.dest('build'));
-  return stream;
 });
 ```
 
-##### Return a promise
+##### Accept a callback
 
-```javascript
-var Q = require('q');
+```js
+var del = require('del');
 
-gulp.task('somename', function() {
-  var deferred = Q.defer();
+gulp.task('clean', function(done) {
+  del(['.build/'], function(err) {
+    done(err)
+  });
+  // or
+  // del(['.build/'], done);
```

Comment:
can we just use this example instead of the extra functions.  keep things more performant in the docs